### ISSUE TITLE
im2rec.py Segmentation fault

### ISSUE THREAD

**Issue by T800GHB** [apache/incubator-mxnet#5407]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:Ubuntu 14.04

Compiler:g++ 4.8.4
               cuda 8.0

Package used (Python/R/Scala/Julia):Python

MXNet version:v0.8

Or if installed from source:from source

MXNet commit hash (`git rev-parse HEAD`):ceb9f0187a31d528e5566f810d933cf4834d3282

If you are using python package, please provide

Python version and distribution:Anaconda3-4.2.0 or Anaconda2-4.2.0

If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.
[16:14:29] src/c_api/../operator/custom-inl.h:66: New registration is overriding existing custom operator custom_op
Segmentation fault (core dumped)

## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.
I follow this tutorial http://mxnet.io/tutorials/python/image_io.html
when i execute this line, error raise
os.system('python %s/tools/im2rec.py --list=1 --recursive=1 --shuffle=1 --test-ratio=0.2 data/caltech data/101_ObjectCategories'%MXNET_HOME)
## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1.
2.
3.

## What have you tried to solve it?

1.
2.
3.

**Comment by T800GHB** [comment]
Install opencv into anaconda will cause this problem.
Install it by: conda install opencv

**Comment by renewjoy** [comment]
I have the same problem when i use im2rec.py,   but use bin/im2rec to product .rec file from .lst file  is OK.

system: Ubuntu 16.04
g++: g++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609

```
py=/opt/anaconda2/bin/python


# 列出所有的图片
$py ~/mxnet/tools/im2rec.py --list True --recursive True car_train train/
~/mxnet/bin/im2rec car_train.lst train/ car_train.rec  resize=256   # it's ok

## below will Segmentation fault
# $py ~/mxnet/tools/im2rec.py --resize 256 --quality 95 --num-thread 1 car_train train/


$py ~/mxnet/tools/im2rec.py --list True --recursive True car_test test/
~/mxnet/bin/im2rec car_test.lst test/ car_test.rec  resize=256   ## it's ok

## below will Segmentation fault
#$py ~/mxnet/tools/im2rec.py --resize 256 --quality 95 --num-thread 1 car_test test/
```

when I use the following command:
```
$py ~/mxnet/tools/im2rec.py --resize 256 --quality 95 --num-thread 1 car_train train/
```

the im2rec.py will `Segmentation fault` in the below line
```
img = cv2.imread(fullpath, args.color)
```

I download opencv3 from url: 
`https://anaconda.org/menpo/opencv3/3.1.0/download/linux-64/opencv3-3.1.0-py27_0.tar.bz2`

and then installed with:
```
/opt/anaconda2/bin/conda install opencv3-3.1.0-py27_0.tar.bz2
```

**Comment by T800GHB** [comment]
@renewjoy I think the key is version of opencv
I have tried, opencv 2.x.x could work perfectly, but opencv 3.x.x will cause many problem

**Comment by austingg** [comment]
make sure your python-opencv and c++-opencv the same version.  you may check like this:
``` shell 
ldd mxent.so | grep opencv 
pip list | grep opencv 
````

**Comment by ysh329** [comment]
Same error: Segmentation fault (core dumped)

**Comment by dusigh** [comment]
I have come across the same segmentation fault error. As @renewjoy mentioned, it happens at the line
`img = cv2.imread(fullpath, args.color)`

**Comment by ysh329** [comment]
Besides, not only `img = cv2.imread(fullpath, args.color)`, but other-any-function calls about `cv2`, which results in this segmentation fault, such as fintuning or predict based on pretrained models according to official tutorials.

When I replace the `cv2` functions with `skimage` lib to load, resize images, it's okay. But, I have to admit, in fact, it's not a permanent solution. Because, in some situation, low-level call of MXNet will also occur this problem, I think it's about `opencv`, but not surely.

After that, I use docker to make data preparing using `im2rec.py`, fine-tuning and prediction. It's okay and smoothly-used.

**Comment by dusigh** [comment]
@ysh329 I actually ran into this problem in a mxnet/python:gpu container. May I ask which docker image you used to get it run smoothly? Thanks.

**Comment by ysh329** [comment]
@dusigh Strange, I used `mxnet/python:gpu` and it's okay. Besides, I also build other images based on this image, [this link](https://hub.docker.com/r/ysh329/mxnet/).

**Comment by dusigh** [comment]
It seems that if I move the line of "import cv2" to above where the mxnet/python path is appended to sys.path, the problem is gone. see below


```
@@ -2,13 +2,13 @@
 from __future__ import print_function
 import os
 import sys
-
+import cv2
 curr_path = os.path.abspath(os.path.dirname(__file__))
 sys.path.append(os.path.join(curr_path, "../python"))
 import mxnet as mx
 import random
 import argparse
-import cv2
 import time
 import traceback
```

**Comment by kevinlin311tw** [comment]
Same here. Just simply import `cv2` before importing `mxnet`. This should be able to solve the problem.
It seems like there could be a few compatibility problems between MXNet and opencv3

**Comment by TheBigGreenBall** [comment]
Yup. I have this issue as well. Even with the simplest lines of code

```
>>> import mxnet as mx
>>> import cv2
>>> img = cv2.imread('10067-False.jpg', 1)
Segmentation fault
```

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
